salt/modules/suse_ip.py

```diff
@@ -0,0 +1,1180 @@
+        bond.update({"primary": opts["primary"]})
+
+    return bond
+
+
+def _parse_settings_bond_6(opts, iface):
+
+    """
+    Filters given options and outputs valid settings for bond6.
+    If an option has a value that is not expected, this
+    function will log what the Interface, Setting and what it was
+    expecting.
+    """
+    bond = {"mode": "6"}
+    bond.update(_parse_settings_miimon(opts, iface))
+
+    if "miimon" not in opts:
+        _raise_error_iface(iface, "miimon", "integer")
+
+    if "primary" in opts:
+        bond.update({"primary": opts["primary"]})
+
+    return bond
+
+
+def _parse_settings_vlan(opts, iface):
+
+    """
+    Filters given options and outputs valid settings for a vlan
+    """
+    vlan = {}
+    if "reorder_hdr" in opts:
+        if opts["reorder_hdr"] in _CONFIG_TRUE + _CONFIG_FALSE:
+            vlan.update({"reorder_hdr": opts["reorder_hdr"]})
+        else:
+            valid = _CONFIG_TRUE + _CONFIG_FALSE
+            _raise_error_iface(iface, "reorder_hdr", valid)
+
+    if "vlan_id" in opts:
+        if opts["vlan_id"] > 0:
+            vlan.update({"vlan_id": opts["vlan_id"]})
+        else:
+            _raise_error_iface(iface, "vlan_id", "Positive integer")
+
+    if "phys_dev" in opts:
+        if len(opts["phys_dev"]) > 0:
+            vlan.update({"phys_dev": opts["phys_dev"]})
+        else:
+            _raise_error_iface(iface, "phys_dev", "Non-empty string")
+
+    return vlan
+
+
+def _parse_settings_eth(opts, iface_type, enabled, iface):
+    """
+    Filters given options and outputs valid settings for a
+    network interface.
+    """
+    result = {"name": iface}
+    if "proto" in opts:
+        valid = [
+            "static",
+            "dhcp",
+            "dhcp4",
+            "dhcp6",
+            "autoip",
+            "dhcp+autoip",
+            "auto6",
+            "6to4",
+            "none",
+        ]
+        if opts["proto"] in valid:
+            result["proto"] = opts["proto"]
+        else:
+            _raise_error_iface(iface, opts["proto"], valid)
+
+    if "mtu" in opts:
+        try:
+            result["mtu"] = int(opts["mtu"])
+        except ValueError:
+            _raise_error_iface(iface, "mtu", ["integer"])
+
+    if "hwaddr" in opts and "macaddr" in opts:
+        msg = "Cannot pass both hwaddr and macaddr. Must use either hwaddr or macaddr"
+        log.error(msg)
+        raise AttributeError(msg)
+
+    if iface_type not in ("bridge",):
+        ethtool = _parse_ethtool_opts(opts, iface)
+        if ethtool:
+            result["ethtool"] = " ".join(
+                ["{} {}".format(x, y) for x, y in ethtool.items()]
+            )
+
+    if iface_type == "slave":
+        result["proto"] = "none"
+
+    if iface_type == "bond":
+        if "mode" not in opts:
+            msg = "Missing required option 'mode'"
+            log.error("%s for bond interface '%s'", msg, iface)
+            raise AttributeError(msg)
```

Comment:
Hi @s0undt3ch. `AttributeError` error here means that there is missing mandatory mode attribute for bonding interface. Most of the code is almost a copy of `rh_ip`, but as there are some important differences in network config betwen RedHat and SUSE based distros, it could be harder to maintain the same code for both. Will check all the suggested changes today. Thanks